docs/conf.py

```diff
@@ -37,18 +38,28 @@
 # https://github.com/sphinx-doc/sphinx/issues/6518#issuecomment-589613836
 
 # When building docs, enable `from __future__ import annotations` everywhere.
-def _rewrite(p):
+def _rewrite(p, add=True):
+  prepend_line = 'from __future__ import annotations\n'
   with open(p) as f:
     contents = f.read()
+  if add and not contents.startswith(prepend_line):
+    contents = prepend_line + contents
+  elif not add and contents.startswith(prepend_line):
+    contents = contents[len(prepend_line):]
+  else:
+    return
   with open(p, 'w') as f:
-    f.write('from __future__ import annotations\n')
     f.write(contents)
 
-if 'READTHEDOCS' in os.environ:
```

Comment:
I'm a bit unhappy that a doc build would touch source files in my git checkout, which is why I guarded this with `READTHEDOCS`. WDYT?

Reply:
I feel the same, this whole thing is a big hack. But when I test the doc building locally, I want to see what will appear in RTD. So, I would run it with READTHEDOCS on. I guess if we leave the conditionals, others can skip it. 

However, I have tested it with a dirty checkout and it seems safe to use. 
An alternative would be to make a copy of the sources and run the building there.

Reply:
I have added back the READTHEDOCS guard, although I think that it is not a useful guard. 
I have also changed the developer.rst to suggest setting this on before running sphinx locally. I assume you want to see the documentation as it will be rendered on readthedocs.